server/src/main/java/org/elasticsearch/action/search/VersionMismatchException.java

```diff
@@ -1,27 +0,0 @@
-/*
- * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
- * or more contributor license agreements. Licensed under the "Elastic License
- * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
- * Public License v 1"; you may not use this file except in compliance with, at
- * your election, the "Elastic License 2.0", the "GNU Affero General Public
- * License v3.0 only", or the "Server Side Public License, v 1".
- */
-
-package org.elasticsearch.action.search;
-
-import org.elasticsearch.ElasticsearchException;
-import org.elasticsearch.common.io.stream.StreamInput;
-
-import java.io.IOException;
-
-public class VersionMismatchException extends ElasticsearchException {
```

Comment:
I'm not sure it's worth keeping this class and its registrations around, for if/when we do need capabilities checks on search queries?

Reply:
I'd say delete it; if/when we need new checks we can introduce a new exception.